app/src/main/java/com/mithridat/nonoconverter/backend/solver/NonogramSolver.java

```diff
@@ -95,7 +110,17 @@ void init() {
      *         false, otherwise
      */
     boolean check() {
-        return true;
+        for (int i = 0; i < _left.length; i++) {
```

Comment:
Is it possible to create one function for this and next loops? You can use foreach in these loops.